test/unit/route-reading.js

```diff
@@ -0,0 +1,85 @@
+/* global afterEach, before, beforeEach, describe, document, expect, inject, it, module, spyOn */
+'use strict';
+
+describe('Reading from $route service', function() {
+  beforeEach(module('angular-google-analytics'));
+  beforeEach(module(function (AnalyticsProvider) {
+    AnalyticsProvider
+      .setAccount('UA-XXXXXX-xx')
+      .logAllCalls(true)
+      .enterTestMode();
+  }));
+  
+  afterEach(inject(function (Analytics) {
+    Analytics.log.length = 0; // clear log
+  }));
+  
+  it('should not activate $route reading', function() {
+    inject(function(Analytics) {
+      expect(Analytics.configuration.readFromRoute).toBe(false);
+    });
+  });
+  
+  describe('without $route service available', function() {
+    beforeEach(module(function(AnalyticsProvider, $provide){
+      AnalyticsProvider.readFromRoute(true);
+      $provide.service('$log', function() {
```

Comment:
With the sinon spy you can remove the $provide.service statement altogether and just tweak the test case.